gordon-plugin/src/main/kotlin/com/banno/gordon/AdbExtensions.kt

```diff
@@ -174,8 +177,23 @@ internal fun List<Device>.reinstall(
                     logger.lifecycle("${device.serialNumber}: installing $instrumentationPackage")
                     device.safeUninstall(installTimeoutMillis, instrumentationPackage)
                     device.installApk(installTimeoutMillis, instrumentationApk).bind()
-                }
+                }.ignoreErrorIfPossible(device, logger, ignoreProblematicDevices, problematicDevices)
             }
         }.awaitAll()
     }.forEach { it.bind() }
 }
+
+internal fun Either<Throwable, Unit>.ignoreErrorIfPossible(
+    device: Device,
+    logger: Logger,
+    ignoreProblematicDevices: Boolean,
+    problematicDevices: MutableList<Device>,
+): Either<Throwable, Unit> =
+    when {
+        this is Either.Left && ignoreProblematicDevices -> {
+            problematicDevices.add(device)
+            logger.lifecycle("${device.serialNumber}: ignored because ${value.message}")
+            Unit.right()
+        }
+        else -> this
+    }
```

Comment:
A couple things here:
- I think this would be a more idiomatic way to write the same thing with Arrow
- I think `logger.warn` would be more appropriate here than `lifecycle`

```suggestion
): Either<Throwable, Unit> = fold(
    ifLeft = {
        Either.conditionally(ignoreProblematicDevices, ifFalse = { it }) {
            problematicDevices.add(device)
            logger.warn("${device.serialNumber}: ignored installation failure", it)
        }
    },
    ifRight = { it.right() }
)
```